appserver/batch/glassfish-batch-connector/src/main/java/fish/payara/jbatch/persistence/rdbms/JBatchJDBCPersistenceManager.java

```diff
@@ -255,2216 +246,2203 @@ public void updateCheckpointData(CheckpointDataKey key, CheckpointData value) {
      */
     @Override
     public void createCheckpointData(CheckpointDataKey key, CheckpointData value) {
-        logger.entering(CLASSNAME, "createCheckpointData", new Object[] { key, value });
+        logger.entering(CLASSNAME, "createCheckpointData", new Object[]{key, value});
 
         insertCheckpointData(key.getCommaSeparatedKey(), value);
 
         logger.exiting(CLASSNAME, "createCheckpointData");
     }
 
     /**
-     * Set the default schema to the username obtained from the connection based on the datasource
+     * Set the default schema to the username obtained from the connection based
+     * on the datasource
      *
      * @return the name of the default database schema
      * @throws SQLException
      */
     public String setDefaultSchema() throws SQLException {
-		logger.finest("Entering setDefaultSchema");
+        logger.finest("Entering setDefaultSchema");
+
+        logger.finest("Java EE mode, getting connection from data source");
+        try (Connection connection = dataSource.getConnection()) {
+
+            DatabaseMetaData dbmd = connection.getMetaData();
+            schema = dbmd.getUserName();
+            String databaseProductName = dbmd.getDatabaseProductName();
+            if (databaseProductName.contains("MySQL")) {
```

Comment:
These `contains` checks were changed from doing lower case (case-less) comparison to case-sensitive ones. I feel this could be a source of error. Is this a intentional change?